### ISSUE TITLE
Multiple NEG/Backend support

### ISSUE THREAD

**Issue by rwkarg** [GoogleCloudPlatform/gke-autoneg-controller#24]
A GKE service can have multiple NEGs created for different ports but there isn't a way to associate a specific NEG with a specific backend.

For example, a service exposes two ports. Port 443 for HTTP2/grpc and port 8443 HTTP1.1 for metrics and diagnostics endpoints. Two NEGs can be created for those two ports but there is no way to associate the port 443 NEG with `backendA` and the port 8443 NEG with `backendB`.

They must be behind separate backends since they use different protocols (`HTTP2` and `HTTP`) to talk from the backend service to the NEG instances.

**Comment by rosmo** [comment]
I created support for multiple backends a bit over a year ago and we are now revisiting the issue. I suggest the following (and I'm very happy to hear any opinions on these):

1. Add a new annotation syntax matching the `cloud.google.com/neg` annotation:
```
cloud.google.com/neg: '{"exposed_ports": {"80":{},"443":{}}}'
controller.autoneg.dev/neg: '{ "endpoint_groups": { "80": { "name": "foo", "max_rate_per_endpoint": 1000 } , "443": { "name": "bar", "max_connections_per_endpoint": 42 } } }'
```
2. We keep the existing annotation style for backwards compatibility for "version 1.0", but emit a deprecation warning.
3. We remove the old-style annotation in "version 1.x".

**Comment by rwkarg** [comment]
I did basically the same thing as @rosmo mentioned in point 1 with the addition of an optional `internal` property (bool) to handle regional backend services on Internal Load Balancers (region pulled from `http://metadata.google.internal/computeMetadata/v1/instance/attributes/cluster-location`).

**Comment by rosmo** [comment]
@rwkarg would you be willing to do a PR?

**Comment by rwkarg** [comment]
I wrote our `not-autoneg` controller in C# so while it was a great learning experience into how Kubernetes controllers work, it's not conducive to a PR.

One thing to keep in mind is that Internal LBs currently have regional backend services. I don't know what the structure would be with potential globally routable Internal LBs (similar to how L4 Internal load balancer can be configured to be globally routable) would look like. A simple `bool` for internal may or may not be appropriate for some future state.

**Comment by soellman** [comment]
Fixed by #34
